### PR TITLE
Fix archive automation Core Lane dispatch

### PR DESCRIPTION
**Pull request by Kbediako**
## What
- Dispatch `core-lane.yml` after archive automation creates or updates an archive PR branch, using the archive PR branch ref and failing closed if dispatch cannot be requested.
- Add `workflow_dispatch` support to Core Lane with archive PR audit inputs, PR-base resolution, and diff-budget override metadata lookup for archive dispatches.
- Grant `actions: write` to the reusable archive workflow and both archive automation callers so `gh workflow run` can create the required check.
- Add focused workflow contract tests plus the CO-356 docs/task packet; refresh the Apr 25 docs-freshness date-boundary rows needed for green gates.

## Why
Archive PR #637 is structurally mergeable but blocked because branch protection requires `Core Lane` and workflow-authored PR updates leave native `pull_request` runs in `action_required` / zero-job state. This preserves branch protection and gives future archive PR heads an explicit required-check path instead of a one-off manual workaround.

## Validation
- `npm run test -- tests/archive-automation-workflow.spec.ts`
- `node scripts/delegation-guard.mjs`
- `node scripts/spec-guard.mjs --dry-run`
- `npm run build`
- `npm run lint` (passes with existing `DelegationMcpHealth.test.ts` warnings)
- `npm run test`
- `npm run docs:check`
- `npm run docs:freshness`
- `npm run repo:stewardship`
- `node scripts/diff-budget.mjs`
- `git diff --check`
- Standalone review: bounded-success, no actionable issues after fixes
- Elegance pass: no simplification beyond the single dispatch seam/test contract

Fixes CO-356.
Follow-up filed separately: CO-362 for the out-of-scope `ARCHIVE_AUTOMERGE_TOKEN` 401 credential repair.

## Summary by CodeRabbit

* **New Features**
  * Archive automation now dispatches the required Core Lane check for archive PR branches, with input validation and fail-closed behaviour to ensure required checks run.

* **Tests**
  * Added workflow contract tests verifying dispatch paths, input validation and archive workflow integration.

* **Documentation**
  * Added spec, PRD, action plan, task checklist and docs-freshness records detailing approach, rollout and validation steps.

### CODE REVIEW

**Comment by Kbediako** [comment]
@codex review

File: tasks/specs/linear-56c2656d-853f-43f9-91af-455970800060-archive-core-lane-checks.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- Add a safe `workflow_dispatch` Core Lane path for archive automation PR heads.
+- Keep branch protection and required context names unchanged.
+- Add focused workflow contract coverage.
+
+## Acceptance
+- `core-lane.yml` supports dispatch with archive PR audit inputs.
+- `archive-automation-base.yml` dispatches Core Lane after a PR is created or updated.
+- Dispatch failure fails the archive automation run.
+- Tests prove the workflow contract.
+
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/specs/linear-56c2656d-853f-43f9-91af-455970800060-archive-core-lane-checks.md:68]: @coderabbitai Fixed in 5d64efe483d056d61c9db1c881d7d9153ca6266a. The mini-spec now includes the required issue-shaping sections, and the latest commit also binds the archive Core Lane dispatch to PR number, branch, head SHA, open main PR state, same repo, same head ref, and same head SHA. Please re-review and resolve.

File: tasks/specs/linear-56c2656d-853f-43f9-91af-455970800060-archive-core-lane-checks.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+- No secret rotation or GitHub branch-protection mutation.
+
+## Parity matrix
+| Surface | Current behavior | Target behavior |
+| --- | --- | --- |
+| Archive PR creation | Workflow `GITHUB_TOKEN` updates leave native `pull_request` runs approval-required or no-job. | Archive workflow explicitly dispatches `core-lane.yml` on the archive PR branch. |
+| Required checks | PR `#637` has no reported `Core Lane` check and cannot satisfy branch protection. | Archive PR heads receive a real `Core Lane` check result before merge. |
+| Diff budget override | Native PR metadata is available only for `pull_request` events. | Archive dispatch resolves PR metadata from the archive PR number before running diff budget. |
+| Auto-merge token | Current token failure is `401 Bad credentials`. | Token repair is handled separately in `CO-362` after the required-check path exists. |
+
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/specs/linear-56c2656d-853f-43f9-91af-455970800060-archive-core-lane-checks.md:56]: @coderabbitai Fixed in 106c3f3dc7d324b81fe48be030667b2ef418f76c. The mini-spec now has an explicit Not done if section covering missing Core Lane results, dispatch failures, one-off-only unblocks, unresolved PR metadata for archive dispatch, and keeping CO-362 scoped away from required-check production. Please re-review and resolve.

File: tasks/specs/linear-56c2656d-853f-43f9-91af-455970800060-archive-core-lane-checks.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+## Not done if
+- Archive PR heads still lack a `Core Lane` result.
+- Dispatch failures do not fail the archive automation run.
+- The fix only unblocks PR `#637` while future archive PRs can reproduce the missing-check state.
+- Diff budget or downstream-scope checks cannot resolve PR metadata from the archive dispatch inputs.
+- The out-of-scope `CO-362` auto-merge token repair becomes a blocker for producing the required `Core Lane` check.
+
+## Parity matrix
+| Surface | Current behavior | Target behavior |
+| --- | --- | --- |
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply tasks/specs/linear-56c2656d-853f-43f9-91af-455970800060-archive-core-lane-checks.md:51]: @coderabbitai Fixed in 812aa79b3eac2fca3db700ca6c24abd92e592835. Added the blank line before the parity table and reran the focused docs guards successfully. Please re-review and resolve.
